### PR TITLE
Remove new line when adding settings

### PR DESCRIPTION
**Pull request by srisankethu**

### CODE REVIEW

File: coalib/output/ConfWriter.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def __write_key_val(self, keys, val):
 
         self.__file.write((self.__key_delimiter + " ").join(keys) + " " +
                           self.__key_value_delimiter + " " + val + "\n")
-        self.__wrote_newline = False
+        self.__wrote_newline = True
```
Comment (sils1297) [inline coalib/output/ConfWriter.py]: maybe we should just not write that newline at all? Wouldn't that be simpler? This adds the newline only in one very specific case which isn't helping a lot anymore

Comment (srisankethu) [inline coalib/output/ConfWriter.py]: If we do that then when a default section does not exist in a .coafile then the file would be
`[Default]
[some_section]` is that ok? They would not be any new line after `[Default]`

**Comment by sils1297** [comment]
yep I think that's fine

File: coalib/output/ConfWriter.py (hunk 1 above)
Comment (srisankethu) [inline coalib/output/ConfWriter.py]: removing ``self.__wrote_newline = True`` is not making any change.

**Comment by srisankethu** [comment]
Sorry! I misinterpreted the functionality earlier. My changes seem fine i guess.

File: coalib/output/ConfWriter.py (hunk 1 above)
Comment (sils1297) [inline coalib/output/ConfWriter.py]: I don't understand this :/

**Comment by sils1297** [comment]
see https://github.com/coala/coala/pull/2797#discussion_r81313327 please, I did https://github.com/coala/coala/pull/2797#issuecomment-250729892 from my mail without the proper context which might be the misunderstanding

File: coalib/output/ConfWriter.py (hunk 1 above)
Comment (srisankethu) [inline coalib/output/ConfWriter.py]: You mean the the line `self.__wrote_newline = True`. I want me to remove that right? Yep it not of any use except for one case. I will update the PR by removing the above line and update the tests.

**Comment by sils1297** [comment]
I would remove  all occurrences of the wrote newline thing whatsoever and
not write the newline in any case.

**Comment by sils1297** [comment]
i.e. remove https://github.com/coala/coala/pull/2797/commits/8c7b8adea2ec9821bde5c6267443fa3395d6cf95#diff-8a58c9fbbbf258ce13b73976719838fbR79

**Comment by sils1297** [comment]
Do you understand why?

**Comment by srisankethu** [comment]
Yep!

**Comment by sils1297** [comment]
ack https://github.com/coala/coala/pull/2797/commits/524f81def6fabdee5755a7c0e311a329491502d1

**Comment by sils1297** [comment]
Can you rebase?

**Comment by srisankethu** [comment]
Ok :+1:

**Comment by sils1297** [comment]
it's not - forgot to fetch?

**Comment by sils1297** [comment]
Hi,

it looks like your PR is out of date and needs a rebase. [This page](http://coala.io/git) may help you to get started on this. (*Do not press the update branch button. It's evil.*)

Add the original coala repository as a remote:

```
git remote add upstream https://github.com/coala-analyzer/coala.git
```

(Be sure to use coala-bears on the coala-bears repository :))

Then go to your branch and rebase it onto our master:

```
git checkout <yourbranch>
git fetch upstream
git rebase --interactive upstream/master
```

Now push it up. You will need to do a force push because you change history here (do not do this on master branches :)):

```
git push --force
```

If you're just looking to fix an issue very quickly and not interested in contributing to coala long term, we can fix it up for you - just tell us :).

Ping us on [our gitter channel](https://coala.io/chat) if you need help!

**Comment by srisankethu** [comment]
Oh, come on! Don't show me that template again. Done :+1:

**Comment by srisankethu** [comment]
I thought since the last commit was 5 days ago, there won't be any changes to the repo.

**Comment by sils1297** [comment]
sometimes showing the template again reminds people of a step they forgot :P

**Comment by sils1297** [comment]
ack https://github.com/coala/coala/pull/2797/commits/173678beefb8c6b2e15eaa122d22556f71e7c8f7

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**